tools/clang/test/CodeGenHLSL/quick-test/groupshared-base-cast.hlsl

```diff
@@ -0,0 +1,41 @@
+// RUN: %dxc -E main -T cs_6_0 %s | FileCheck %s
+
+// CHECK: @[[gs0:.+]] = addrspace(3) global i32 undef
```

Comment:
Can you add a comment saying what this is testing and why you need all of inheritance, groupshared, rwbyteaddressbuffers and barriers in order to test it?

Reply:
I'll add a comment for what it's testing, however the barrier and output buffer may be necessary for this to not have groupshared use optimized away, be considered dead code, or have a race condition detected.  It could be more minimally tested with -fcgl or maybe -Od, but it might be more fragile to codegen changes we might want to make as well.